### PR TITLE
Completely translate documentation to English

### PR DESCRIPTION
**Pull request by muscache**
This PR adds English translations for the current Chinese documentation
Documentation will be divided by locale. eg. `docs/en`, `docs/zh`

### CODE REVIEW

**Comment by Newbytee** [comment]
Did you just edit the built HTML files manually, or did you rebuild using Sphinx?

File: docs/en/html/Components/BLE/ble_stack/ble_stack.html [hunk 1]
```diff
@@ -0,0 +1,971 @@
...
+    if (!window.google || !google.translate ||
+        !google.translate.TranslateElement) {
+      if (!window[cbId]) {
+        window[cbId] = function() {
+          window[teId] = newElem();
+          show();
+        };
+      }
+      var s = document.createElement('script');
+      s.src = 'https://translate.google.com/translate_a/element.js?cb=' +
```
Comment (Newbytee) [inline docs/en/html/Components/BLE/ble_stack/ble_stack.html:75]: Are you just running the text through Google Translate?

**Comment by muscache** [comment]
> Did you just edit the built HTML files manually, or did you rebuild using Sphinx?
I edited them manually. It would be great if you could point me to using Sphinx!

**Comment by Newbytee** [comment]
> > Did you just edit the built HTML files manually, or did you rebuild using Sphinx?
> I edited them manually. It would be great if you could point me to using Sphinx!

My pull request builds this using Sphinx: https://github.com/pine64/bl_iot_sdk/pull/5 (repo: https://github.com/Newbytee/bl_iot_sdk/tree/initial-eng-docs). If you're familiar with git you can base your pull request on that. You can check the docs/en/source directory for the .rst files that the HTML is built from.

File: docs/en/html/Components/BLE/ble_stack/ble_stack.html (hunk 1 above)
Comment (muscache) [reply docs/en/html/Components/BLE/ble_stack/ble_stack.html:75]: Yes, all the docs are translated via google. I feel it’s a good starting point till we get a linguist who knows Chinese and English!

If you’re curious about how I did it, I used the Google Translate extension for the job. To work, it injects scripts into the document; I cleaned up using my own little script (which unfortunately missed a tag in that document)

**Comment by muscache** [comment]
> > > Did you just edit the built HTML files manually, or did you rebuild using Sphinx?
> > > I edited them manually. It would be great if you could point me to using Sphinx!
> 
> My pull request builds this using Sphinx: #5 (repo: https://github.com/Newbytee/bl_iot_sdk/tree/initial-eng-docs). If you're familiar with git you can base your pull request on that. You can check the docs/en/source directory for the .rst files that the HTML is built from.

Thanks for the quick reply. I’ll write a quick script to convert the HTML into .rst files and PR tomorrow

File: docs/en/html/Components/BLE/ble_stack/ble_stack.html (hunk 1 above)
Comment (Newbytee) [reply docs/en/html/Components/BLE/ble_stack/ble_stack.html:75]: I'm not so convinced this is a good idea. I discussed the translation I got from Google Translate when translating the Hello world example with a Chinese friend and it seems a fair bit of detail was lost in translation, even with something simple like that.

Comment (muscache) [reply docs/en/html/Components/BLE/ble_stack/ble_stack.html:75]: Until we get someone who can translate between chinese and english, GTranslate is our only option

**Comment by gamelaster** [comment]
Closed in favor of pine64/ble602-docs#1 .

**Pull request closed without merging**